examples/step-95/step-95.cc

```diff
@@ -0,0 +1,1711 @@
+      return is_inside(face_category.first) && is_inside(face_category.second);
+    }
+
+    inline bool
+    is_mixed_face(std::pair<unsigned int, unsigned int> face_category) const
+    {
+      return (is_inside(face_category.first) &&
+              is_intersected(face_category.second)) ||
+             (is_intersected(face_category.first) &&
+              is_inside(face_category.second));
+    }
+
+    inline bool is_intersected_face(
+      std::pair<unsigned int, unsigned int> face_category) const
+    {
+      return is_intersected(face_category.first) &&
+             is_intersected(face_category.second);
+    }
+
+    // Helper function to determine the relevant cell lengths of a face batch;
+    VectorizedArrayType
+    compute_diameter_of_inner_face_batch(unsigned int face_batch_index) const
+    {
+      const auto &face_info = matrix_free->get_face_info(face_batch_index);
+
+      VectorizedArrayType diameter = 0.;
+      for (unsigned int v = 0;
+           v < matrix_free->n_active_entries_per_face_batch(face_batch_index);
+           ++v)
+        {
+          const auto cell_batch_index_interior =
+            face_info.cells_interior[v] / n_lanes;
+          const auto cell_lane_index_interior =
+            face_info.cells_interior[v] % n_lanes;
+          const auto cell_batch_index_exterior =
+            face_info.cells_exterior[v] / n_lanes;
+          const auto cell_lane_index_exterior =
+            face_info.cells_exterior[v] % n_lanes;
+
+          diameter[v] = std::max(
+            cell_diameter[cell_batch_index_interior][cell_lane_index_interior],
+            cell_diameter[cell_batch_index_exterior][cell_lane_index_exterior]);
+        }
+
+      return diameter;
+    }
+
+    // Helper function which computes the interior penalty parameter for the
+    // SIPG flux.
+    VectorizedArrayType compute_interior_penalty_parameter(
+      const VectorizedArrayType &diameter) const
+    {
+      return 5. / diameter;
+    }
+
+    // ObserverPointer of the MatrixFree and the NonMatching::MappingInfo
+    // objects.
+    ObserverPointer<const MatrixFree<dim, Number, VectorizedArrayType>>
+      matrix_free;
+    ObserverPointer<
+      const NonMatching::MappingInfo<dim, dim, VectorizedArrayType>>
+      mapping_info_cell;
+    ObserverPointer<
+      const NonMatching::MappingInfo<dim, dim, VectorizedArrayType>>
+      mapping_info_surface;
+    ObserverPointer<
+      const NonMatching::MappingInfo<dim, dim, VectorizedArrayType>>
+      mapping_info_faces;
+
+    std::unique_ptr<GenericCellIntegrator> evaluator_cell;
+    std::unique_ptr<GenericCellIntegrator> evaluator_surface;
+    std::unique_ptr<GenericFaceIntegrator> evaluator_face_m;
+    std::unique_ptr<GenericFaceIntegrator> evaluator_face_p;
+
+    AlignedVector<VectorizedArrayType> cell_diameter;
+
+    const unsigned int dof_index  = 0;
+    const unsigned int quad_index = 0;
+
+    bool is_dg = false;
+  };
+
+  // @sect3{The Jacobi preconditioner}
```

Comment:
After this long class, I think it would make sense to put 3 empty lines, not just a single one.